src/views/Pools/components/CakeVaultCard/RecentCakeProfitRow.tsx

```diff
@@ -1,26 +1,44 @@
 import React from 'react'
-import BigNumber from 'bignumber.js'
 import { Flex, Text } from '@pancakeswap/uikit'
-import { useTranslation } from 'contexts/Localization'
 import { useWeb3React } from '@web3-react/core'
+import { useTranslation } from 'contexts/Localization'
+import { useGetApiPrice } from 'state/hooks'
+import useGetVaultUserInfo from 'hooks/cakeVault/useGetVaultUserInfo'
+import useGetVaultPublicData from 'hooks/cakeVault/useGetVaultPublicData'
+import { getBalanceNumber } from 'utils/formatBalance'
+import { getCakeAddress } from 'utils/addressHelpers'
+import { convertSharesToCake } from 'views/Pools/helpers'
 import RecentCakeProfitBalance from './RecentCakeProfitBalance'
 
-interface RecentCakeProfitRowProps {
-  cakeAtLastUserAction: BigNumber
-  userShares: BigNumber
-}
-
-const RecentCakeProfitCountdownRow: React.FC<RecentCakeProfitRowProps> = ({ cakeAtLastUserAction, userShares }) => {
+const RecentCakeProfitCountdownRow = () => {
   const { t } = useTranslation()
   const { account } = useWeb3React()
+  const { cakeAtLastUserAction, userShares, lastUserActionTime } = useGetVaultUserInfo()
   const shouldDisplayCakeProfit =
     account && cakeAtLastUserAction && cakeAtLastUserAction.gt(0) && userShares && userShares.gt(0)
 
+  const { pricePerFullShare } = useGetVaultPublicData()
+  const currentSharesAsCake = convertSharesToCake(userShares, pricePerFullShare)
+  const cakeProfit = currentSharesAsCake.cakeAsBigNumber.minus(cakeAtLastUserAction)
+  const cakeToDisplay = cakeProfit.gte(0) ? getBalanceNumber(cakeProfit, 18) : 0
+
+  const cakePrice = useGetApiPrice(getCakeAddress())
+  const dollarValueOfCake = cakeProfit.times(cakePrice)
+  const dollarValueToDisplay = dollarValueOfCake.gte(0) ? getBalanceNumber(dollarValueOfCake, 18) : 0
+
+  const lastActionInMs = lastUserActionTime && parseInt(lastUserActionTime) * 1000
+  const dateTimeLastAction = new Date(lastActionInMs)
+  const dateStringToDisplay = dateTimeLastAction.toUTCString()
```

Comment:
I'd be interested to see what users in different timezones see when when `toUTCString()` is called. Please check the preview and let me know?

<img width="231" alt="Screenshot 2021-05-07 at 15 31 20" src="https://user-images.githubusercontent.com/79279477/117465150-4c268b80-af49-11eb-99c4-7f96b00c06cc.png">

Reply:
Since it just shows GMT time I think timezone doesn't matter. `.toLocaleString()` can be called on Date and return local time I guess (or just formatted to local format)

![Screenshot 2021-05-11 at 8 59 38](https://user-images.githubusercontent.com/81824236/117765824-407fe100-b237-11eb-9a47-86ad0ab37f18.png)

Reply:
Nice, cheers Cheems that's tidier.